### DISCUSSION TITLE
Support for std::env::vars in nodejs target

### DISCUSSION THREAD

**Discussion by Thesephi** [wasm-bindgen/wasm-bindgen#2890] [Ideas]
Hi wasm-bindgen team & community,

I stumbled upon this use case just now:

- I have a function in `lib.rs` which iterates over the list of environment variables using `std::env::vars()`
- I compile successfully (with `wasm-pack build --target nodejs`)
- I import the generated wasm module into my Node.js script, everything runs fine except it panics as soon as it reaches the `std::env::vars()` part:

  ![wasm-pack-nodejs-unsupported-std-env-vars](https://user-images.githubusercontent.com/2597375/167273952-55712f12-ef18-4919-b240-670cea32259d.png)

To reproduce on your local, you could:
- checkout my example project at this tag: https://github.com/dklabco/cosmos-tools/tree/discussion-std-env-vars-1
- execute `make build`, then `make std-env-vars`
- you should now see the panicking happens as I [elaborate here](https://github.com/dklabco/cosmos-tools/blob/discussion-std-env-vars-1/examples/_lab_/std-env-vars.mjs#L3-L4)
  - and this is the [line of rust code](https://github.com/dklabco/cosmos-tools/blob/discussion-std-env-vars-1/src/lib.rs#L63) that causes the panic

My current setups (in case it helps):
- rustc 1.60.0 (7737e0b5c 2022-04-04)
- wasm-pack 0.10.2
- Node.js v18.1.0

I tried looking up in the Discord channel, the GitHub Issues and Discussions sections, but I wasn't able to find a relevant thread. Could I have overlooked anything? Otherwise, is this "use case" justified? Is there anything worth discussing or anything I could elaborate on / help with?

Thank you in advance for your feedback & insights!

**Comment by ranile** [comment]
This has to do with the implementation in std. The `nodejs` flag only specifies the JS loader that is generated

You can import the node JS function for reading environment variables and call it. That should give you the environment variables
